src/Type/Php/ArrayKeysFunctionDynamicReturnTypeExtension.php

```diff
@@ -36,7 +37,16 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 			return $this->phpVersion->arrayFunctionsReturnNullWithNonArray() ? new NullType() : new NeverType();
 		}
 
-		return $arrayType->getKeysArray();
+		$searchValueType = isset($functionCall->getArgs()[1]) ? $scope->getType($functionCall->getArgs()[1]->value) : null;
+		$strictType = isset($functionCall->getArgs()[2]) ? $scope->getType($functionCall->getArgs()[2]->value) : null;
+		$strict = $strictType instanceof ConstantBooleanType && $strictType->getValue();
```

Comment:
instaneceof *Type is rarely correct, and in this case it isn't :) You also need to account for mixed or bool being passed in here (so we cannot be sure if it's strict or not), so as always, isSuperTypeOf() wins :)

Reply:
adapted, I think it should be more correct now? I still exit early the same way later though